services/cd-service/pkg/service/commit_deployment.go

```diff
@@ -95,13 +96,53 @@ func getCommitDeploymentInfoForApp(ctx context.Context, h *db.DBHandler, commit,
 	if err != nil {
 		return nil, fmt.Errorf("Could not get all environments from all_environments metadata: %v", err)
 	}
+
+	for _, app := range apps {
+		commitDeploymentStatusForApp, err := getCommitDeploymentInfoForApp(ctx, s.DBHandler, releaseNumber, app, allEnvironments)
```

Comment:
hmm, instead of spawning a query to the db for each app, maybe we should have a `get CommitDeploymentInfoForApps` that does it for all of them.
SQL would look something like:
```
SELECT ... FROM all_environments GROUP BY app ORDER BY version
```

Reply:
We would have to query the database for each app to get the release version for each environment, and I don't see a way to do this in one query.

Reply:
Resolved.
Now all apps and their releases are returned in a single query.